controllers/front/ProductController.php

```diff
@@ -1027,7 +1027,8 @@ public function getTemplateVarProduct()
         $product['minimal_quantity'] = $this->getProductMinimalQuantity($product);
         $product['quantity_wanted'] = $this->getRequiredQuantity($product);
         $product['extraContent'] = $extraContentFinder->addParams(['product' => $this->product])->present();
-        $product['ecotax'] = Tools::convertPrice((float) $product['ecotax'], $this->context->currency, true, $this->context);
+        $ecotax_rate = (float) Tax::getProductEcotaxRate();
+        $product['ecotax'] = Tools::ps_round((float) $product['ecotax'] * (1 + $ecotax_rate / 100), 6, null);
```

Comment:
```suggestion
        $product['ecotax'] = Tools::ps_round((float) $product['ecotax'] * (1 + $ecotax_rate / 100), $this->context->currency->precision, null);
```

Reply:
I will change the approach of my PR ... Thank you  @Progi1984!